### PR TITLE
fix: Warning when dragging document between collections with different user permissions

### LINKED ISSUE #2150: Warning when dragging document between collections with different user permissions
When moving a document today it just happens immediately, It would be nice if when moving a document from a collection with limited access to another with different access we would show a confirmation dialog that the user intended to change the permission level.

### PR DESCRIPTION
**Pull request by iamsaumya**
closes #2150

### CODE REVIEW

File: app/components/Sidebar/components/CollectionLink.js [hunk 1]
```diff
@@ -67,9 +77,19 @@ function CollectionLink({
...
     drop: (item, monitor) => {
+      const { id, permission } = item;
       if (monitor.didDrop()) return;
       if (!collection) return;
-      documents.move(item.id, collection.id);
+      if (collection.id === item.collectionId) return;
+      if (
+        permission !== collection.permission &&
+        (permission === null || collection.permission === null)
+      ) {
```
Comment (tommoor) [inline app/components/Sidebar/components/CollectionLink.js:87]: I think we need to make this so the modal is only triggered when the new permission is more open than the old permission, eg. only in the case of moving from `no default access` -> `read default access`.

File: app/components/Sidebar/components/CollectionLink.js [hunk 2]
```diff
@@ -203,6 +223,17 @@ function CollectionLink({
             index={index}
           />
         ))}
+      <Modal
+        title={t("Document Move")}
```
Comment (tommoor) [inline app/components/Sidebar/components/CollectionLink.js:227]:
```suggestion
        title={t("Move document")}
```

File: app/components/Sidebar/components/CollectionLink.js [hunk 3]
```diff
@@ -203,6 +223,17 @@ function CollectionLink({
...
+        onRequestClose={() => handlePermissionClose()}
```
Comment (tommoor) [inline app/components/Sidebar/components/CollectionLink.js:228]:
```suggestion
        onRequestClose={handlePermissionClose}
```

File: app/components/Sidebar/components/CollectionLink.js [hunk 4]
```diff
@@ -203,6 +223,17 @@ function CollectionLink({
...
+        isOpen={permissionOpen}
+      >
+        <DocumentReparent
+          item={itemRef.current}
+          collection={collection}
+          onSubmit={() => handlePermissionClose()}
```
Comment (tommoor) [inline app/components/Sidebar/components/CollectionLink.js:234]:
```suggestion
          onSubmit={handlePermissionClose}
```

File: app/components/Sidebar/components/DocumentLink.js [hunk 5]
```diff
@@ -128,7 +128,14 @@ function DocumentLink(
...
   const [{ isDragging }, drag] = useDrag({
     type: "document",
-    item: () => ({ ...node, depth, active: isActiveDocument }),
+    item: () => ({
+      ...node,
+      depth,
+      active: isActiveDocument,
+      permission: collection?.permission,
+      collectionId: collection?.id,
+      collectionName: collection?.name,
```
Comment (tommoor) [inline app/components/Sidebar/components/DocumentLink.js:137]: We should store only the `id` here, if the name and other details are needed in the modal then they can be loaded from the collections store?

**Review by tommoor — COMMENTED** [review]
Thank you

**Comment by iamsaumya** [comment]
Ohh, I didn't know it is a conflict here 😅

**Review by tommoor — APPROVED** [review]

**Pull request merged**